[stable/falco] Add Falco chart

nestorsalceda opened the pull request:
<!--
Thank you for contributing to kubernetes/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/kubernetes/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/kubernetes/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the
history. This will make it easier to identify new changes. The PR will be squashed
anyways when it is merged. Thanks.

For fast feedback, please @-mention maintainers that are listed in the Chart.yaml file.

Please make sure you test your changes before you push them. Once pushed, a CircleCI
will run across your changes and do some initial checks and linting. These checks run
very quickly. Please check the results. We would like these checks to pass before we
even continue reviewing your changes.
-->

**What this PR does / why we need it**:
This PR adds a chart for installing Sysdig Falco in Kubernetes. 
                                                                                                                                                                        Sysdig Falco is a behavioral activity monitor designed to detect anomalous activity in your applications. 

You can use Falco to monitor run-time security of your Kubernetes applications and internal components.                                    

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

None, because this is a new chart.

**Special notes for your reviewer**:

Commit:
[stable/falco] Add Falco chart

Commit:
Fix indentation and other stuff reported by CI

Commit:
Add appVersion to Chart.yaml

nestorsalceda commented:
/check-cla

nestorsalceda commented:
/assign @lachie83

Commit:
Specify container resources

Commit:
Allow to load external Falco rules

Commit:
Move GCSCC integrations to a top level integrations section

We can correlate falco.* keys for falco related settings, and refer them
in Falco Wiki

Commit:
Rename deployment to fakeEventGenerator

First one is too generic

Commit:
Add OWNERS file

unguiculus commented:
Please have a look at our RBAC best practices.

https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/rbac.md

unguiculus commented:
/assign

Commit:
Separate rbac and serviceAccount

Follow RBAC best practices: https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/rbac.md

Commit:
Use falco.serviceAccount name template for cluster role binding

unguiculus commented on stable/falco/README.md:
```diff
@@ -50,7 +50,8 @@ The following table lists the configurable parameters of the Falco chart and the
 | `resources`                                     | Specify container resources                                         | `{}`                                                                                   |
 | `rbac.create`                                   | If true, create & use RBAC resources                                | `true`                                                                                 |
-| `rbac.serviceAccountName`                       | If rbac.create is false, use this value as serviceAccountName       | `default`                                                                              |
+| `serviceAccount.create`                         | Create serviceAccount                                               | `default`                                                                              |
```
default -> true

unguiculus commented on stable/falco/templates/clusterrolebinding.yaml:
```diff
@@ -0,0 +1,20 @@
+subjects:
+  - kind: ServiceAccount
+    name: {{ template "falco.serviceAccountName" .}}
+    namespace: default
```
default -> {{ .Release.Namespace }}

unguiculus commented on stable/falco/templates/clusterrolebinding.yaml:
```diff
@@ -0,0 +1,20 @@
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: {{ template "falco.fullname" .}}
+  namespace: default
```
Remove namespace

unguiculus commented on stable/falco/templates/configmap-rules.yaml:
```diff
@@ -0,0 +1,10 @@
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+data: {}
```
Why do you create an empty configmap. The above rules file is not used by the chart. Resources created by Helm should not be edited manually as you suggest in the readme. This would get you in trouble when updating a release.

nestorsalceda replied:
Because we want to allow that chart users adds its own rules. Let me to explain a bit more the context. 

Although Falco comes with a nice default rule set for detecting weird behavior in containers, our users are going to customize the run-time security rule sets or policies for the specific container images and applications they run. In Falco we provide an /etc/falco/rules.d directory for dropping new files with rules and we want to allow our users to upload its rules to that directory, so our idea was to mount the /etc/falco/rules.d directory as a configMap and allow users to add rules to that configMap.

For us, the ideal situation would be allowing to pass rules files when installing the helm but 
I have been reading about this problem https://github.com/kubernetes/helm/issues/3276 or in [Accessing files inside templates](https://github.com/kubernetes/helm/blob/master/docs/chart_template_guide/accessing_files.md#accessing-files-inside-templates) and looks like this is not supported. Perhaps you have some tips or guidance or a workaround for this issue?

Thanks!

unguiculus replied:
If you use Helm you should use it exclusively. Manipulating file installed by Helm will get you in trouble. Why don't you make the configmap configurable via `values.yaml`?

nestorsalceda replied:
Done! We load custom rules via values.yaml file :)

Thanks!

Commit:
Fixes required from reviewer

Commit:
Allow passing rules in an external file instead of editing configMap by hand

lachie83 commented on stable/falco/Chart.yaml:
```diff
@@ -0,0 +1,19 @@
+apiVersion: v1
+name: falco
+version: '0.1'
```
Remove single quotes

Commit:
Remove quotes from Chart version

I'm not sure if this break lint stage in CircleCI

lachie83 committed:
Update Chart.yaml
